Bug 1182225 release notes redesign

sgarrity opened the pull request:
This replaces PR https://github.com/mozilla/bedrock/pull/3581 from a shared branch as there are some more updates to push.

For some context, there was some initial work by @yglazko, then the bulk of the work was done by @malena. I have added a few (as yet unsquashed, for easier review) clean up commits at the end.

sgarrity commented on bedrock/firefox/templates/firefox/releases/aurora-notes.html:
```diff
@@ -4,16 +4,31 @@
+  <div id="nav" class="navigator">
+    <div class="container">
+      <ul class="menu">
+        <li><a href="{{ url('firefox') + 'notes/' }}">{{ _('Desktop') }}</a></li>
```
@pmclanahan or @jgmize Is there a better way to reference these links? I'm not sure if there's a way within the `url()` function to references the various types of release notes pages.

schalkneethling commented on bedrock/firefox/templates/firefox/releases/release-notes.html:
```diff
@@ -8,164 +8,311 @@
+              </ul>
+            </div>
+            {% endif %}
+          {% endfor %}
```
Nit: Fix indentation above.

schalkneethling commented on media/css/firefox/releasenotes-firefox.less:
```diff
@@ -0,0 +1,564 @@
+.mdn-icon:before {
+    content: " ";
+    background-image: url('/media/img/firefox/releasenotes/mdn-icon.png');
+    background-size: 100%;
```
Use `.background-size()` mixin

schalkneethling commented on media/css/firefox/releasenotes-firefox.less:
```diff
@@ -0,0 +1,564 @@
+    border-left: 10px solid transparent;
+    border-right: 10px solid transparent;
+    border-bottom: 10px solid #fff;
+    content: "";
```
Nit: single quotes for consistency

schalkneethling commented on media/css/firefox/releasenotes-firefox.less:
```diff
@@ -0,0 +1,564 @@
+    content: "\00A0\f164\00A0";
+    font-family: 'Font Awesome';
+}
+#Known::before {
```
For all of the above use `:` as apposed to `::` for wider browser support.

schalkneethling commented on media/css/firefox/releasenotes-firefox.less:
```diff
@@ -0,0 +1,564 @@
+    font-family: 'Font Awesome';
+}
+#Known::before {
+    content: "\00A0\f05e\00A0";
```
Nit: Single quotes for consistency

schalkneethling commented on media/css/firefox/releasenotes-firefox.less:
```diff
@@ -0,0 +1,564 @@
+    }
+    .notes-head {
+        .intro {
+            margin: 20px 0;
```
Nit: `margin: @baseLine 0;`

schalkneethling commented:
@sgarrity If I go to for example:
`http://localhost:8000/en-US/firefox/new/`

The "What's New" link is:
`http://localhost:8000/en-US/firefox/notes/`

when I click on "What's New", I am sent to the following page:
`http://localhost:8000/en-US/firefox/42.0/releasenotes/`

Which is a 404. I do not see any additions or changes to any `urls.py` files, is that correct? Is there a known reason why I am getting a 404 in the above scenario? Thanks!

schalkneethling commented:
There is a few redirects here:
https://github.com/mozilla/bedrock/blob/master/bedrock/firefox/redirects.py#L53

Perhaps they should no longer be?

craigcook commented on bedrock/firefox/templates/firefox/releases/release-notes.html:
```diff
@@ -8,164 +8,311 @@
+          {% for note in new_features if note.tag == "New" %}
+            {% if loop.first %}
+            <div class="section-wrapper">
+              <p id="{{ note.tag }}">{{ note.tag }}</p>
```
This should probably be a heading rather than a paragraph. It acts as a title for the list that follows.

I'd also also convert the ID to lowercase (I think it's just `lower(note.tag)`?) and update the corresponding CSS rules (since CSS selectors are case-senstive). I'm not sure how written-in-stone these tags are but making sure the IDs are lowercase would ensure some consistency.

sgarrity replied:
> I'd also also convert the ID to lowercase (I think it's just lower(note.tag)?) and update the corresponding CSS rules (since CSS selectors are case-sensitive).

Good idea. For the record, it was `note.tag|lower()` rather than `lower(note.tag)`.

craigcook commented on bedrock/firefox/templates/firefox/releases/release-notes.html:
```diff
@@ -8,164 +8,311 @@
+          {% for note in new_features if note.tag == "Fixed" %}
+            {% if loop.first %}
+            <div class="section-wrapper">
+              <p id="{{ note.tag }}">{{ note.tag }}</p>
```
Same thing here; heading with a lowercase ID. Though come to think of it, if the only purpose for these IDs is to identify the section so we can link to it, it should be on the section's container and not the heading.

craigcook commented on bedrock/firefox/templates/firefox/releases/release-notes.html:
```diff
@@ -8,164 +8,311 @@
-        </ul>
-      </aside>
-    {% endif %}
+      <div id="nav" class="navigator">
```
Should probably be a `nav` element.

craigcook commented on bedrock/firefox/templates/firefox/releases/release-notes.html:
```diff
@@ -8,164 +8,311 @@
+          {% if release.product == 'Firefox' %}
+            <li class="current">{{ _('Desktop') }}</li>
+            <li><a href="{{ url('firefox') + 'android/notes/' }}">{{ _('Android') }}</a></li>
+            <li><a href="{{ url('firefox') + 'ios/1.1/releasenotes' }}">{{ _('iOS') }}</a></li>
```
Should the version number be hard-coded here? iOS notes are still taking shape but it might be better to have a generic URL with a redirect (so the redirect could be updated in one place for new releases), rather than hard-code it into the links (and have to update them in lots of places for each release).

sgarrity replied:
> Should the version number be hard-coded here? iOS notes are still taking shape but it might be better to have a generic URL with a redirect (so the redirect could be updated in one place for new releases), rather than hard-code it into the links (and have to update them in lots of places for each release).

Good question. I'm not sure how best to address this though. Maybe product-details can tell be the #. I'll check.

schalkneethling replied:
Is there a resolution to this?

sgarrity replied:
I've asked on IRC, but I'll repeat here since people are travelling:

@pmac or @jgmize: Is there a way I can link from a template to the latest version of the release notes for Firefox for iOS (or any release channel/platform combo)? I see how the url()s are defined here, but I'm not sure if/how I can reference them from a template: https://github.com/mozilla/bedrock/blob/master/bedrock/firefox/urls.py#L116-L133

sgarrity replied:
@schalkneethling There's a redirect bug (https://bugzilla.mozilla.org/show_bug.cgi?id=1231189) keeping us from being able to link to the latest iOS release notes. I propose we leave the hard-coded URLs in place until that bug is fixed. I could file another bug to follow-up/clean-up.

I have update the other URLs to use the bedrock helpers though.

sgarrity commented:
> @sgarrity If I go to for example:
> http://localhost:8000/en-US/firefox/new/
> 
> The "What's New" link is:
> http://localhost:8000/en-US/firefox/notes/
> 
> when I click on "What's New", I am sent to the following page:
> http://localhost:8000/en-US/firefox/42.0/releasenotes/
> 
> Which is a 404. I do not see any additions or changes to any urls.py files, is that correct? Is there a known reason why I am getting a 404 in the above scenario? Thanks!

@schalkneethling I can't replicate this locally. The http://localhost:8000/en-US/firefox/42.0/releasenotes/ URL isn't a 404 for me. Are you up-to-date with `bin/sync_all`?

sgarrity commented:
Thanks for the feedback from the happiest place on earth! Most of the feedback is addressed (or has a comment/question. Ready for another round of review!

schalkneethling commented:
@sgarrity Aha! I thought it might be something I am missing. Will sync all the things.

schalkneethling commented:
At mobile viewport sizes, the labels bleeds into the description section.

![screen shot 2015-12-07 at 11 55 55](https://cloud.githubusercontent.com/assets/10350960/11633491/958ff91a-9cd9-11e5-84fc-fd108b7a0d2d.png)

schalkneethling commented:
Maybe it is just me but, having the "Congrats! You’re using the latest version of Firefox" at the bottom seems kinda odd. Any specific reason this is not at the top of the page?

schalkneethling commented:
Could the `new`, `fixed`, `chaged` etc. labels be a larger font size? Seems very small to me at 2048x1280

schalkneethling commented on bedrock/firefox/templates/firefox/includes/release-menu.html:
```diff
@@ -0,0 +1,11 @@
+<nav id="nav-main" role="navigation">
+  <span id="nav-main-toggle" class="toggle" role="button" aria-controls="nav-main-menu" aria-expanded="false" tabindex="0">{{_('Menu')}}</span>
+  <ul id="nav-main-menu">
+    <li class="first"><a class="more" rel="external" href="https://developer.mozilla.org">{{_('MDN')}}</a>
```
Is there a reason why only the MDN link has `class="more"`?

sgarrity replied:
> Is there a reason why only the MDN link has class="more"?

This was also before my time. I presume this is to help indicate that the MDN link is "external". The semantics of the class name aren't great, but that isn't really user-visible.

sgarrity commented:
> At mobile viewport sizes, the labels bleeds into the description section.

Fixed! Good catch.


> Maybe it is just me but, having the "Congrats! You’re using the latest version of Firefox" at the bottom seems kinda odd. Any specific reason this is not at the top of the page?

Yeah, this is a bit odd. I presume the logic behind this is that they wanted a download button available in the footer, and showing a download button to up-to-date visitors is redundant, hence this phrase.


> Could the new, fixed, chaged etc. labels be a larger font size? Seems very small to me at 2048x1280

I may be taking more liberties than I can afford with the design, but I agree and went ahead with this change. Shouldn't be too controversial as it now uses stock Sandstone h4 styles.

sgarrity commented:
@schalkneethling Thanks again for the review. Should be ready to look at again.

schalkneethling commented:
@sgarrity Small thing, should we override the default color for tabzilla here and make the foreground color `#333`? Hard to read at the moment.

sgarrity commented:
> @sgarrity Small thing, should we override the default color for tabzilla here and make the foreground color #333? Hard to read at the moment.

Good idea. Done.

schalkneethling commented on bedrock/firefox/templates/firefox/releases/aurora-notes.html:
```diff
@@ -4,16 +4,31 @@
-    {% if release.major_version()|int() < 35 %}
-	{{ super() }}
-    {% endif %}
+  <div id="nav" class="navigator">
```
Nit: `nav` element here?

schalkneethling commented on bedrock/firefox/templates/firefox/releases/beta-notes.html:
```diff
@@ -5,3 +5,26 @@
 {% set channel_name = 'Beta' %}
+
+{% block platform_switch %}
+  <div id="nav" class="navigator">
```
Nit: As before, should we instead use a `nav` element here?

sgarrity commented:
@schalkneethling Updated those remaining `nav` elements.

schalkneethling commented on bedrock/firefox/templates/firefox/releases/dev-browser-notes.html:
```diff
@@ -1,18 +1,17 @@
+    <p>{{ _('{date}')|f(date=release.release_date|l10n_format_date) }}</p>
+  </div>
+  <div class="description">
+      <h2>{{ _('Version {version}, first offered to Firefox Developer Edition channel users on {date}')|f(channel=release.channel, date=release.release_date|l10n_format_date, version=release.version) }}</h2>
```
Nit: Perhaps it is better to use another `h4` here as apposed to a `h2`? Looking at the semantics and display, it looks very much like it shares a lot with the `h4` above on line 10.

schalkneethling replied:
Looking at the outline, perhaps we can change line 9 and 10 to be a `h2` and `h3` respectively? We then have a heading flow of:

```
1
-- 2
---- 3
-- 2
```
Thoughts?

schalkneethling commented on bedrock/firefox/templates/firefox/releases/release-notes.html:
```diff
@@ -8,164 +8,311 @@
+    <div class="col">
+      <h2>{{ _('Need help or want to leave feedback?') }}</h2>
+      <a rel="external" href="https://bugzilla.mozilla.org/enter_bug.cgi">{{ _('File a bug') }}</a>
+      <a rel="external" href="https://input.mozilla.org/en-US/feedback?utm_source=releasenotes">{{ _('Leave feedback') }}</a>
```
Probably best to not "hard-code" `en-US` in the URL.

schalkneethling commented on media/css/firefox/releasenotes-firefox.less:
```diff
@@ -0,0 +1,576 @@
+.mdn-icon:before {
+    content: ' ';
+    background-image: url('/media/img/firefox/releasenotes/mdn-icon.png');
+    .background-size(100%);
```
use `.background-size(100%);` mixin

schalkneethling commented on media/css/firefox/releasenotes-firefox.less:
```diff
@@ -0,0 +1,576 @@
+        margin: 0;
+    }
+    h4 {
+        .font-size(20px);
```
`.font-size(@baseLine);`?

schalkneethling commented on media/css/firefox/releasenotes-firefox.less:
```diff
@@ -0,0 +1,576 @@
+
+    li {
+        display: inline-block;
+        padding: 20px 16px;
```
`padding: @baseLine 16px;`?

schalkneethling commented on media/css/firefox/releasenotes-firefox.less:
```diff
@@ -0,0 +1,576 @@
+        .border-box();
+        border-bottom: solid 1px #eee;
+        display: table;
+        margin: auto auto;
```
Nit: This could probably just be `margin: auto;`

schalkneethling commented on media/css/firefox/releasenotes-firefox.less:
```diff
@@ -0,0 +1,576 @@
+    li {
+        display: list-item;
+        padding: 15px;
+        margin-left: 5px 0px 0px 30px;
```
No units when value is `0`

schalkneethling commented on media/css/firefox/releasenotes-firefox.less:
```diff
@@ -0,0 +1,576 @@
+    }
+    .notes-section {
+        .section-wrapper {
+            padding: 20px 0;
```
`20px` => `@baseLine`

schalkneethling commented on media/css/firefox/releasenotes-firefox.less:
```diff
@@ -0,0 +1,576 @@
+
+.notes-footer {
+    .container {
+        padding: 40px;
```
`(@baseLine * 2)`?

schalkneethling commented on media/css/firefox/releasenotes-firefox.less:
```diff
@@ -0,0 +1,576 @@
+    background-color: #fff;
+    box-shadow: 0px 1px 1px 0px rgba(50, 50, 50, 0.4);
+    display: none;
+    margin-top: 20px;
```
`20px` => `@baseLine`

schalkneethling commented:
A couple of small things here and there, but this is looking good :dancers:

sgarrity commented:
> A couple of small things here and there, but this is looking good :dancers:

@schalkneethling I've cleaned the remaining items, leaving a few '20px' references that probably aren't directly related to the @baseLine var.

I've got an open question about the iOS release notes version number link.

schalkneethling commented:
r+ :notes: Thanks @sgarrity

jbertsch commented:
Hi-
@m-alexis and I discussed this with Jon, Alex, and Josh, plus Sylvestre, today, and we want to wait to push to prod until Wednesday, Dec 16 minimize risk.
Thanks for all the hard work on this one,
Jen

schalkneethling commented:
@jbertsch I guess this is safe to merge now?

sgarrity commented:
This has been rebased and squashed.
